src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -282,7 +282,7 @@ export default function SavedObjectFactory(esAdmin, kbnIndex, Promise, Private,
           if (_.get(err, 'origError.status') === 409) {
             const confirmMessage = `Are you sure you want to overwrite ${this.title}?`;
 
-            return safeConfirm(confirmMessage)
+            return confirmModalPromise(confirmMessage, { confirmButtonText: 'Overwrite' })
```

Comment:
Is it possible to get rid of `confirmModalPromise` entirely and just use `confirmModal` instead? The reason I ask is because it seems a bit confusing to have two ways of doing the same thing, only with vey slightly different interfaces. Are there use cases where it's better to use one over the other?